Wikipedia/Code/WMFCVLInfo.m

```diff
@@ -0,0 +1,337 @@
+    [self.sections enumerateObjectsUsingBlock:block];
+}
+
+- (void)enumerateColumnsWithBlock:(nonnull void(^)(WMFCVLColumn * _Nonnull column, NSUInteger idx, BOOL * _Nonnull stop))block {
+    [self.columns enumerateObjectsUsingBlock:block];
+}
+
+- (nullable WMFCVLAttributes *)layoutAttributesForItemAtIndexPath:(NSIndexPath *)indexPath {
+    NSInteger sectionIndex = indexPath.section;
+    
+    if (sectionIndex < 0 || sectionIndex >= self.sections.count) {
+        return nil;
+    }
+    
+    WMFCVLSection *section = self.sections[sectionIndex];
+    NSInteger itemIndex = indexPath.item;
+    if (itemIndex < 0 || itemIndex >= section.items.count) {
+        return nil;
+    }
+    
+    WMFCVLAttributes *attributes = section.items[itemIndex];
+    assert(attributes != nil);
+    return attributes;
+}
+
+- (nullable WMFCVLAttributes *)layoutAttributesForSupplementaryViewOfKind:(NSString *)elementKind atIndexPath:(NSIndexPath *)indexPath {
+    NSInteger sectionIndex = indexPath.section;
+    if (sectionIndex < 0 || sectionIndex >= self.sections.count) {
+        return nil;
+    }
+    
+    WMFCVLSection *section = self.sections[sectionIndex];
+    
+    WMFCVLAttributes *attributes = nil;
+    if ([elementKind isEqualToString:UICollectionElementKindSectionHeader]) {
+        NSInteger itemIndex = indexPath.item;
+        if (itemIndex < 0 || itemIndex >= section.headers.count) {
+            return nil;
+        }
+        attributes = section.headers[itemIndex];
+    } else if ([elementKind isEqualToString:UICollectionElementKindSectionFooter]) {
+        NSInteger itemIndex = indexPath.item;
+        if (itemIndex < 0 || itemIndex >= section.footers.count) {
+            return nil;
+        }
+        attributes = section.footers[itemIndex];
+    }
+    
+    assert(attributes != nil);
+    return attributes;
+}
+
+- (void)updateContentSize {
+    __block CGSize newSize = self.boundsSize;
+    newSize.height = 0;
+    [self enumerateColumnsWithBlock:^(WMFCVLColumn * _Nonnull column, NSUInteger idx, BOOL * _Nonnull stop) {
+        CGFloat columnHeight = column.height;
+        if (columnHeight > newSize.height) {
+            newSize.height = columnHeight;
+        }
+    }];
+    self.contentSize = newSize;
+}
+
+- (void)updateContentSizeWithInvalidationContext:(WMFCVLInvalidationContext *)context {
+    CGSize oldContentSize = self.contentSize;
+    
+    [self updateContentSize];
+    
+    CGSize contentSizeAdjustment = CGSizeMake(self.contentSize.width - oldContentSize.width, self.contentSize.height - oldContentSize.height);
+    context.contentSizeAdjustment = contentSizeAdjustment;
+}
+
+- (void)updateWithInvalidationContext:(nonnull WMFCVLInvalidationContext *)context delegate:(id <WMFColumnarCollectionViewLayoutDelegate>)delegate collectionView:(UICollectionView *)collectionView {
+    if (context.boundsDidChange) {
```

Comment:
Is this semantically being used the same way as `invalidateEverything`? If so, I guess you needed this because `invalidateEverything` is readonly?